src/FileFormats/MPS/MPS.jl

```diff
@@ -1287,7 +1287,10 @@ function _add_variable(model, data, variable_map, i, name)
     variable_map[name] = x
     MOI.set(model, MOI.VariableName(), x, name)
     set = bounds_to_set(data.col_lower[i], data.col_upper[i])
-    if set !== nothing
+    if set isa MOI.Interval
+        MOI.add_constraint(model, x, MOI.GreaterThan(set.lower::Float64))
+        MOI.add_constraint(model, x, MOI.LessThan(set.upper::Float64))
+    elseif set !== nothing
         MOI.add_constraint(model, x, set)
     end
```

Comment:
Sorry if this is too nit-picky, but: Could creating the "helper" `Interval` set be an overhead for large files?

Something like this would be possible here (but makes it more convoluted):

```julia
if isfinite(data.col_lower[i])
    MOI.add_constraint(model, x, MOI.GreaterThan(data.col_lower[i]::Float64))
end

if isfinite(data.col_upper[i])
    MOI.add_constraint(model, x, MOI.LessThan(data.col_upper[i]::Float64))
end
```

This would also handle two edge cases differently:

1. If `data.col_lower[i] > data.col_upper[i]`, it adds the bounds keeping the model infeasible (`bounds_to_set` would currently remove the bounds and create a free variable)
2. If `data.col_lower[i] == data.col_upper[i]`, it sticks to adding bounds, and does not convert those into a single equality

(1.) is maybe not really important, but (2.) could lead to similar problems as the initial version: 

- I create a variable that somewhere along the way gets equal bounds, then
- write that to a file, read it back in, and
- the bounds are gone, `has_*_bound` will return `false`, and I can't change the bounds anymore.

Further, this also impacts how `fix(x, value; force=false)` behaves. With bounds it errors, with the `EqualTo` it overwrites.

Reply:
> Could creating the "helper" Interval set be an overhead for large files?

Nope. It's a `struct` that is on the stack.

> Something like this would be possible here

`bounds_to_set` already does this :smile: it will return an `Interval` only if both sides are finite.

> the bounds are gone, has_*_bound will return false, and I can't change the bounds anymore.

You'd need to `unfix(x); set_lower_bound(x, lb)`.

The issue is that JuMP doesn't offer a nice way to interact with interval bounds.

I agree that lower/upper/fixed bounds are more convoluted than a plain lower/upper bound. You'll need to write some JuMP code to deal with the different situations.